### PR TITLE
Update Inovelli VZM35-SN fan switch for firmware 1.05

### PR DESCRIPTION
**Pull request by Rogue136198**
## Proposed change

Updating this to support changes made to the latest beta firmware v1.05 for VZM35SN

## Additional information

This is my first time I have submitted an update to a quirk. If someone could please advise if this change should be done this way 
or if another way would be better such as adding a new quirk specifically for firmware 1.05 in the event this breaks those devices 
still using 1.04

## Checklist

- [x] The changes are tested and work correctly
- [ ] `pre-commit` checks pass / the code has been formatted using Black
- [ ] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by javicalle** [comment]
Definitely that would be a breaking change for the not updated devices.
IMHO it will be better to put the new device signature in another quirk.
Just put your code in a new class (ie: `InovelliVZM35SN_var02(CustomDevice)`), keeping the current quirk as is.

**Comment by codyhackw** [comment]
We'll also need to update the \_\_init__.py file for the new attributes, I believe this is what's going to be needed in total - 

<details><summary>Add to VZM35SN.py with the import you added for the fan cluster</summary>

```
class InovelliVZM35SNv5(CustomDevice):
    """VZM35-SN Fan Switch"""
    signature = {
        MODELS_INFO: [("Inovelli", "VZM35-SN")],
        ENDPOINTS: {
            1: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: DeviceType.DIMMABLE_LIGHT,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    Fan.cluster_id,
                    Diagnostic.cluster_id,
                    INOVELLI_VZM35SN_CLUSTER_ID,
                    WWAH_CLUSTER_ID,
                ],
                OUTPUT_CLUSTERS: [
                    Ota.cluster_id,
                ],
            },
            2: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                ],
                OUTPUT_CLUSTERS: [
                    Identify.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                ],
            },
            3: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                ],
                OUTPUT_CLUSTERS: [
                    Identify.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                ],
            },
            242: {
                PROFILE_ID: zgp.PROFILE_ID,
                DEVICE_TYPE: zgp.DeviceType.PROXY_BASIC,
                INPUT_CLUSTERS: [],
                OUTPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
            },
        },
    }

    replacement = {
        ENDPOINTS: {
            1: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: DeviceType.DIMMABLE_LIGHT,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    Fan.cluster_id,
                    Diagnostic.cluster_id,
                    Inovelli_VZM35SN_Cluster,
                    WWAH_CLUSTER_ID,
                ],
                OUTPUT_CLUSTERS: [
                    Ota.cluster_id,
                ],
            },
            2: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                ],
                OUTPUT_CLUSTERS: [
                    Identify.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    Inovelli_VZM35SN_Cluster,
                ],
            },
            3: {
                PROFILE_ID: zha.PROFILE_ID,
                DEVICE_TYPE: DeviceType.DIMMER_SWITCH,
                INPUT_CLUSTERS: [
                    Basic.cluster_id,
                    Identify.cluster_id,
                    Groups.cluster_id,
                    Scenes.cluster_id,
                ],
                OUTPUT_CLUSTERS: [
                    Identify.cluster_id,
                    OnOff.cluster_id,
                    LevelControl.cluster_id,
                    Inovelli_VZM35SN_Cluster,
                ],
            },
            242: {
                PROFILE_ID: zgp.PROFILE_ID,
                DEVICE_TYPE: zgp.DeviceType.PROXY_BASIC,
                INPUT_CLUSTERS: [],
                OUTPUT_CLUSTERS: [GreenPowerProxy.cluster_id],
            },
        },
    }
```

</details>

<details><summary>Update VZM35SN Cluster in __init__.py</summary>

```
class Inovelli_VZM35SN_Cluster(Inovelli_VZM31SN_Cluster):
    """Inovelli VZM35-SN custom cluster."""

    attributes = {
        key: Inovelli_VZM31SN_Cluster.attributes[key]
        for key in Inovelli_VZM31SN_Cluster.attributes
        if key not in VZM35SN_REMOVES
    }
    attributes.update(
        {
            0x0017: ("quick_start_time", t.uint8_t, True),
            0x001E: ("non_neutral_aux_med_gear_learn_value", t.uint8_t, True),
            0x001F: ("non_neutral_aux_low_gear_learn_value", t.uint8_t, True),
            0x0034: ("smart_fan_mode", t.Bool, True),
            0x0078: ("fan_single_tap_behavior", t.uint8_t, True),
            0x0079: ("fan_timer_display", t.Bool, True),
            0x0081: ("fan_breeze_mode", t.uint32_t, True),
            0x0082: ("fan_module_binding_control", t.uint8_t, True),
            0x0083: ("low_for_bound_control", t.uint8_t, True),
            0x0084: ("medium_for_bound_control", t.uint8_t, True),
            0x0085: ("high_for_bound_control", t.uint8_t, True),
            0x0086: ("led_color_for_bound_control", t.uint8_t, True),
            0x0107: ("smart_fan_led_display_levels", t.uint8_t, True),
        }
    )
```

</details>

File: zhaquirks/inovelli/VZM35SN.py [hunk 1]
```diff
@@ -45,6 +46,7 @@ class InovelliVZM35SN(CustomDevice):
                     Scenes.cluster_id,
                     OnOff.cluster_id,
                     LevelControl.cluster_id,
+                    Fan.cluster_id,
```
Comment (owen2) [inline zhaquirks/inovelli/VZM35SN.py:49]: Does DEVICE_TYPE need to change as well?

**Comment by TheJulianJES** [comment]
Thanks @Rogue136198 and @codyhackw!
@codyhackw I've now added the new quirk class separately and added the custom attributes like you mentioned.

Let me know if the PR looks good to you know.

**Comment by codyhackw** [comment]
@TheJulianJES that looks good to me!

**Comment by TheJulianJES** [comment]
Thanks

**Comment by Rogue136198** [comment]
Whoops, forgot I was working on this. Been a crazy week.

I imported the updates files as a custom quirk and everything seems to be working well. 

Thanks for finishing this up for me.

**Pull request merged**